.github/workflows/main.yml

```diff
@@ -155,7 +163,7 @@ jobs:
   tests-unit:
     runs-on: ubuntu-latest
     needs: build
```

Comment:
I'm fairly certain you'll need this to be

```yml
needs: [changes, build]
```

in order to reference the outputs.